packages/flutter_tools/lib/src/version.dart

```diff
@@ -242,10 +242,8 @@ class FlutterVersion {
         _flutterGit,
       ]);
       await _run(<String>['git', 'fetch', _versionCheckRemote, branch]);
```

Comment:
Since we are fetching the actual latest version from Flutter's Github repository, the version may be different than what is shown when running `flutter upgrade --verify-only`, which displays the version of the upstream as configured by the user (which may not be the same as `_flutterGit` url).

I'm indecisive about what to do about this.

Reply:
Mm, yeah, this not an ideal situation. Independent of this PR, it may be worth reconsidering how we treat `flutter upgrade` upstreams.